src/biogeochem/CNPhenologyMod.F90

```diff
@@ -372,6 +457,31 @@ subroutine CNPhenologyInit(bounds)
 
     if ( use_crop ) call CropPhenologyInit(bounds)
 
+    ! Error checking for parameters
+    if ( (critical_daylight_method == critical_daylight_depends_on_lat) .or. &
+         (critical_daylight_method == critical_daylight_depends_on_veg) .or.  &
+         (critical_daylight_method == critical_daylight_depends_on_latnveg) )then
+        if ( params_inst%crit_dayl_at_high_lat < params_inst%crit_dayl )then
+            call endrun(msg="ERROR crit_dayl_at_high_lat should be higher than crit_dayl on paramsfile"//errmsg(sourcefile, __LINE__))
+        end if
+        if ( params_inst%crit_dayl_at_high_lat >= secspday )then
+            call endrun(msg="ERROR crit_dayl_at_high_lat can NOT be higher than seconds in a day"//errmsg(sourcefile, __LINE__))
+        end if
+        if ( params_inst%crit_dayl >= secspday )then
+            call endrun(msg="ERROR crit_dayl can NOT be higher than seconds in a day"//errmsg(sourcefile, __LINE__))
+        end if
+    end if
+    if ( (critical_daylight_method == critical_daylight_depends_on_lat) .or. &
+         (critical_daylight_method == critical_daylight_depends_on_latnveg) )then
+        if ( params_inst%crit_dayl_lat_slope <= 0.0_r8 )then
+            call endrun(msg="ERROR crit_dayl_lat_slope can not be negative or zero"//errmsg(sourcefile, __LINE__))
+        end if
+        if ( params_inst%crit_dayl_lat_slope >= (secspday - params_inst%crit_dayl_at_high_lat)/ &
+                                                 (90._r8 - critical_offset_high_lat) )then
```

Comment:
I don't understand "can not higher than allowing a day at the poles"

Reply:
OK got it.